.github/workflows/ci.yml

```diff
@@ -64,11 +64,25 @@ jobs:
     name: Verify doc-ui only PRs
     uses: ./.github/workflows/verify_changes.yml
 
+  checkout-ref:
+    name: Checkout head or merge commit
+    runs-on: ubuntu-latest
+    outputs:
+      checkout-ref: ${{ steps.checkout-ref-output.outputs.checkout-ref }}
+    steps:
+      - if: ${{ !contains(github.event.pull_request.labels.*.name, 'checkout-head') }}
+        run: echo "CHECKOUT_REF=${{ github.base_ref }}" >> "$GITHUB_ENV"
+      - if: ${{ contains(github.event.pull_request.labels.*.name, 'checkout-head') }}
+        run: echo "CHECKOUT_REF=${{ github.event.pull_request.head.sha }}" >> "$GITHUB_ENV"
+      - id: checkout-ref-output
+        run: echo "checkout-ref=${{ env.CHECKOUT_REF }}" >> "$GITHUB_OUTPUT"
```

Comment:
Could we move this logic into the `Setup` job above? Each job runs on a separate runner, so we can save ourselves acquiring a `ubuntu-latest` runner each CI run if it's part of the setup logic 😄

Reply:
Good call! Done